src/url.js

```diff
@@ -92,9 +92,14 @@ export function getWinOrigin(win) {
  * testing by freezing the object.
  * @param {string} url
  * @param {boolean=} opt_nocache
+ *   Cache is always ignored on ESM builds, see https://go.amp.dev/pr/31594
  * @return {!Location}
  */
 export function parseUrlDeprecated(url, opt_nocache) {
+  if (IS_ESM) {
+    return /** @type {!Location} */ (new URL(url));
```

Comment:
Oh wow, I just noticed that this won't handle non-absolute URLs, eg `/test.html` or `foo.jpg`. We need to add `location.href`:

```suggestion
    return /** @type {!Location} */ (new URL(url, location.href));
```

 Can we add a test case?

Reply:
Done!

Reply:
I had to revert this to using an `<a>` wth empty `href`. 

When using srcdoc-defined iframes (like on our testing iframes, posisbly on ads), [`location.href` is different from what `a.href` would be](https://iframe-location-srcdoc.glitch.me). This throws an `Invalid URL` error (`new URL('foo', 'about:srcdoc")`)

There's a certain risk to break documents loaded in iframes in general, so I thought it was safer to use `a.href = ''`.